modules/key/main.tf

```diff
@@ -0,0 +1,16 @@
+#########################################################################################
```

Comment:
Can we combine instance creation and key attach as one module by providing bind argument which will be input from user whether he wants to bind a key to instance or not.

Based on user input we can decide whether to invoke both (instance creation and key attach) modules have to be invoked or one should be ok. 

We can discuss this point offline if we want.